### ISSUE TITLE
Old templates relating to current system indices should automatically be deleted

### ISSUE THREAD

**Issue by bhavyarm** [elastic/elasticsearch#80853]
**Elasticsearch version** (`bin/elasticsearch --version`): 7.16 BC5

**OS version** (`uname -a` if on a Unix-like system): darwin

**Description of the problem including expected versus actual behavior**: While upgrading from 6.8.20->7.12.1->7.16.0 UA throws a critical deprecation for Custom mapping types with this message - 
```
Update or remove the following index templates before upgrading to 8.0: [.ml-config, .ml-meta, .ml-notifications, .monitoring-alerts, .triggered_watches, .watch-history-9, .watches, kibana_index_template:.kibana, logstash-index-template, security-index-template]. See https://ela.st/es-deprecation-7-removal-of-types for alternatives to mapping types.
```

After discussion in stack-upgrade it was determined that these system templates need to be removed to user doesn't see them in Kibana

<img width="1792" alt="Screen Shot 2021-11-18 at 1 39 57 PM" src="https://user-images.githubusercontent.com/7074629/142484720-7b493817-e5de-40c7-9129-dde1240dce09.png">

**Comment by bhavyarm** [comment]
cc @gwbrown @droberts195 @williamrandolph

**Comment by jakelandis** [comment]
In the short term @masseyke will open a PR to drop the severity of this warning down to WARN.
We will look to fix/remove these internally used index templates so that they will no longer appear in this warning.
Once that work is done, we will evaluate raising the level back to CRITICAL to raise visibility of user templates that will not work in 8.x.

@romseygeek - Can you help confirm that index templates with custom mapping types are harmless in 8.x if they are never used ?

**Comment by cjcenizal** [comment]
CC @elastic/kibana-stack-management for visibility

**Comment by romseygeek** [comment]
The mappings in a composable template are only parsed by a mapper service if they're actually used, so a bad template that is not referenced anywhere shouldn't cause any problems.

**Comment by rudolf** [comment]
@romseygeek These aren't composable templates though, they're legacy templates created in 6.x

**Comment by romseygeek** [comment]
@rudolf The same applies, they don't get parsed into Mappings unless they're referenced by an index creation request.

**Comment by droberts195** [comment]
For the ML templates I've merged #80874 into 7.16 which will mean that ML legacy templates will disappear very quickly from 7.16 clusters.

Other Elasticsearch plugins could follow the same approach.  If you already have rolling upgrade tests that have existed since 6.8 and cause your internal indices to be created in the old cluster then you can very easily add a test that reproduces the problem by adding code like this to the upgraded cluster step of your rolling upgrade test on the 7.16 branch: https://github.com/elastic/elasticsearch/blob/9d99587ffe646cf55e398cb30acf16ee0a6f84ab/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/MlMappingsUpgradeIT.java#L179-L200

I added that test in the first commit of #80874 and it failed as expected.  Now it passes with the ML code fixed to delete obsolete legacy indices.

Maybe there's time to do something similar for Watcher and Security on Monday so the problem is greatly reduced by the time of 7.16.0 release?

**Comment by mattc58** [comment]
We are closing this for inactivity but will reopen if necessary. Also, we're not sure if deleting older templates would always be a safe operation for all clusters due to BWC concerns.
